edit.go

```diff
@@ -88,43 +107,98 @@ func (w *Client) getPage(pageIDorName string, isName bool) (content string, time
 		"continue":     "",
 	}
 
-	if isName {
-		p.Set("titles", pageIDorName)
-	} else {
-		p.Set("pageids", pageIDorName)
+	for _, identifier := range pageIDsOrNames {
+		if areNames {
+			p.Add("titles", identifier)
+		} else {
+			p.Add("pageids", identifier)
+		}
 	}
 
 	resp, err := w.Get(p)
 	if err != nil {
-		return "", "", err
+		return pages, err
```

Comment:
Why return the empty `pages` map instead of `nil` here (and other similar places in this method)?